File: 002.rb
# By considering the terms in the Fibonacci sequence whose values do not 
# exceed four million, find the sum of the even-valued terms.

LIMIT = 4000000
fibonacci1 = 1
fibonacci2 = 2
sum = 0
while fibonacci2 <= LIMIT
    fibonacci1, fibonacci2 = fibonacci2, fibonacci2 + fibonacci1
    sum += fibonacci1 if fibonacci1.even?
end
puts sum

File: clojure/src/project_euler_clojure/002.clj
;; By considering the terms in the Fibonacci sequence whose values do not
;; exceed four million, find the sum of the even-valued terms.

(ns project-euler-clojure.002)

(defn fib [a b]
  (->> (fib b (+ a b))
       (cons a)
       (lazy-seq)))

(defn problem-002 []
  (->> (take-while (partial >= 4000000) (fib 1 1))
       (filter even?)
       (reduce +)))